applications/serial_lte_modem/src/gnss/slm_at_gnss.c

```diff
@@ -520,16 +519,20 @@ static void fix_rep_wk(struct k_work *work)
 	} else {
 		/* Report to nRF Cloud by best-effort */
 		if (nrf_cloud_ready && location_signify) {
-			err = date_time_now(&ts_ms);
+			struct nrf_cloud_gnss_data gnss = {
+				.type = NRF_CLOUD_GNSS_TYPE_MODEM_NMEA,
+				.mdm_nmea = &nmea
+			};
+
+			err = date_time_now(&gnss.ts_ms);
 			if (err) {
-				err = nrf_cloud_rest_send_location(&rest_ctx, device_id,
-								nmea.nmea_str, -1);
-			} else if (last_ts_ms == 0 ||
-				ts_ms > last_ts_ms + REST_LOCATION_REPORT_MS) {
-				last_ts_ms = ts_ms;
-				err = nrf_cloud_rest_send_location(&rest_ctx, device_id,
-								nmea.nmea_str, ts_ms);
+				gnss.ts_ms = -1;
+			} else if ((last_ts_ms == 0) ||
+				   (gnss.ts_ms > (last_ts_ms + REST_LOCATION_REPORT_MS))) {
+				last_ts_ms = gnss.ts_ms;
 			}
```

Comment:
Original logic is to reduce REST send by minimal interval of REST_LOCATION_REPORT_MS, which is mainly to address GPS continuous mode. New code has dropped the original logic.

Reply:
Oops.  That was not intentional 🙈

Reply:
fixed.
also, is there a reason SLM is using REST instead of just sending the location via MQTT?

Reply:
using REST in this way results in much higher data usage.
here is a commit which removes REST and send the NMEA data via the existing MQTT connection:
https://github.com/nrfconnect/sdk-nrf/commit/fd88368ea7a34a575724d088c5eb4666564fa116
would you like me to add this commit to this PR?

Reply:
The only reason was to avoid JSON encoding in SLM.
If it's too costly in data size, pls do as you proposed, i.e. send by nrf_cloud_send().
Thanks!

Reply:
updated, please review.